service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AgentPersonMapper.java

```diff
@@ -17,40 +17,58 @@
 
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 @Component
-public class PractitionerAgentPersonMapper {
+public class AgentPersonMapper {
 
     private static final Mustache AGENT_STATEMENT_TEMPLATE = TemplateUtils
         .loadTemplate("ehr_agent_person_template.mustache");
     private static final String UNKNOWN = "Unknown";
     private final MessageContext messageContext;
-    private final OrganizationToAgentMapper organizationToAgentMapper;
 
-    public String mapPractitionerToAgentPerson(Practitioner practitioner, Optional<PractitionerRole> practitionerRole,
-        Optional<Organization> organization) {
+    public String mapAgentPerson(AgentDirectory.AgentKey agentKey, String agentDirectoryId) {
         var builder = PractitionerAgentPersonMapperParameters.builder()
-            .practitionerId(messageContext.getIdMapper().getOrNew(ResourceType.Practitioner, practitioner.getIdElement()));
+            .agentId(agentDirectoryId);
 
-        buildPractitionerPrefix(practitioner).ifPresent(builder::practitionerPrefix);
-        var practitionerGiven = buildPractitionerGivenName(practitioner);
-        var practitionerFamily = buildPractitionerFamilyName(practitioner);
+        Optional<Practitioner> practitioner = messageContext.getInputBundleHolder()
+            .getResource(new IdType(agentKey.getPractitionerReference()))
+            .map(resource -> (Practitioner) resource);
```

Comment:
I find the cast method clearer, personally

```suggestion
            .map(Practitioner.class::cast);
```

What do you think?